commit 3fb93c5f46cc857ddeec97b1e178e4d9b83f39ab
Author: Tim Foster <tim.foster@joyent.com>
Date:   2019-03-11T16:45:32+00:00 (7 months ago)
    
    TOOLS-2199 engbld should check for uid == 0 and set $(PFEXEC) accordingly

diff --git a/tools/mk/Makefile.defs b/tools/mk/Makefile.defs
index 8bd6ac9..d8cce8e 100644
--- a/tools/mk/Makefile.defs
+++ b/tools/mk/Makefile.defs
@@ -200,3 +200,16 @@ BUILDIMAGE_MF           ?= \
     "version": "$(BUILDIMAGE_VERSION)",\
     "tags": {"smartdc_service": true}\
     }
+
+#
+# Crude detection of whether we're likely to need to use
+# pfexec to gain root-like privileges. Note that this means
+# $(PFEXEC) will only be defined if we're running the build
+# as non-root.
+#
+UID = $(shell /usr/bin/id -u)
+ifeq ($(UID), 0)
+    PFEXEC=
+else
+    PFEXEC=/usr/bin/pfexec
+endif
diff --git a/tools/mk/Makefile.node_prebuilt.defs b/tools/mk/Makefile.node_prebuilt.defs
index fd4aca4..443bc72 100644
--- a/tools/mk/Makefile.node_prebuilt.defs
+++ b/tools/mk/Makefile.node_prebuilt.defs
@@ -124,7 +124,7 @@ DISTCLEAN_FILES	+= $(NODE_INSTALL) \
 	$(BUILD)/prebuilt-node-* $(BUILD)/prebuilt-npm-*
 
 NODE_PREBUILT_BRANCH ?= master
-NODE_PREBUILT_IMAGE ?= $(shell pfexec mdata-get sdc:image_uuid)
+NODE_PREBUILT_IMAGE ?= $(shell $(PFEXEC) mdata-get sdc:image_uuid)
 ifeq ($(NODE_PREBUILT_TAG),)
 	NODE_PREBUILT_NAME := $(NODE_PREBUILT_VERSION)-$(NODE_PREBUILT_IMAGE)
 else
diff --git a/tools/mk/Makefile.targ b/tools/mk/Makefile.targ
index 33a33a4..159e095 100644
--- a/tools/mk/Makefile.targ
+++ b/tools/mk/Makefile.targ
@@ -389,7 +389,7 @@ prepublish:
 	@echo "Generating $(ENGBLD_BITS_DIR)/$(NAME)/build-environment"
 	@if [[ "$$(uname -s)" == "SunOS" ]]; then \
 		MAKE_PID=$$(/usr/bin/ps -o ppid= -p $$$$); \
-		BUILD_IMG_UUID=$$(/usr/bin/pfexec /usr/sbin/mdata-get sdc:image_uuid); \
+		BUILD_IMG_UUID=$$($(PFEXEC) /usr/sbin/mdata-get sdc:image_uuid); \
 		echo hostname: $$(/usr/bin/hostname) > $(ENGBLD_BITS_DIR)/$(NAME)/build-environment; \
 		/usr/bin/cat /etc/pkgsrc_version >> $(ENGBLD_BITS_DIR)/$(NAME)/build-environment; \
 		echo image_uuid: $$BUILD_IMG_UUID >> $(ENGBLD_BITS_DIR)/$(NAME)/build-environment; \
@@ -441,9 +441,9 @@ buildimage: stamp-buildimage-prep release $(AGENTS:%=%-prebuilt)
 	mkdir -p $(BUILDIMAGE_STAGEDIR)
 	cd $(BUILDIMAGE_STAGEDIR) ; \
 	$(TOP)/deps/eng/tools/buildimage-extract-pkg.sh $(BUILDIMAGE_PKG)
-	pfexec chown -R root:root $(BUILDIMAGE_STAGEDIR)
+	$(PFEXEC) chown -R root:root $(BUILDIMAGE_STAGEDIR)
 	cd $(BUILDIMAGE_STAGEDIR); \
-	pfexec $(BUILDIMAGE) \
+	$(PFEXEC) $(BUILDIMAGE) \
 		-i $(BASE_IMAGE_UUID) \
 		-d $(BUILDIMAGE_STAGEDIR)/root \
 		-m '$(BUILDIMAGE_MF)' \
@@ -451,9 +451,9 @@ buildimage: stamp-buildimage-prep release $(AGENTS:%=%-prebuilt)
 		-P $(NAME)-zfs
 	cp /tmp/$(NAME)-zfs-$(BUILDIMAGE_VERSION).zfs.gz $(ENGBLD_BITS_DIR)/$(NAME)
 	cp /tmp/$(NAME)-zfs-$(BUILDIMAGE_VERSION).imgmanifest $(ENGBLD_BITS_DIR)/$(NAME)
-	pfexec rm /tmp/$(NAME)-zfs-$(BUILDIMAGE_VERSION).zfs.gz
-	pfexec rm /tmp/$(NAME)-zfs-$(BUILDIMAGE_VERSION).imgmanifest
-	pfexec rm -rf $(BUILDIMAGE_STAGEDIR)
+	$(PFEXEC) rm /tmp/$(NAME)-zfs-$(BUILDIMAGE_VERSION).zfs.gz
+	$(PFEXEC) rm /tmp/$(NAME)-zfs-$(BUILDIMAGE_VERSION).imgmanifest
+	$(PFEXEC) rm -rf $(BUILDIMAGE_STAGEDIR)
 
 #
 # Upload the build products from this component's ./bits directory
